src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -1975,7 +1975,13 @@ Klass* JavaThread::security_get_caller_class(int depth) {
 // Returns true if sleep time elapsed as expected, and false
 // if the thread was interrupted.
 bool JavaThread::sleep(jlong millis) {
+  return sleep(millis, 0);
+}
+
+bool JavaThread::sleep(jlong millis, jint nanos) {
```

Comment:
You don't need the overloads at this level - the incoming call should always have millis and nanos, even if nanos is zero.

Reply:
I think the distinction between millis and nanos makes our life unnecessarily hard here. `jlong nanos` is already quite big for practical uses, and so we can simplify this code by only passing `nanos` around. In new revision, I still do two overloads to avoid "unit" mistakes.